### PR TITLE
Added optional server name for the SNI TLS extension

### PR DESCRIPTION
**Pull request by sebbarg**
Server Name Indication (SNI) is an extension to TLS by which a client indicates which hostname it is attempting to connect to at the start of the handshaking process. This allows a server to present multiple certificates on the same IP address.

https://en.wikipedia.org/wiki/Server_Name_Indication

### CODE REVIEW

**Comment by mcollina** [comment]
Can you please add a unit test?

**Comment by sebbarg** [comment]
Hi Matteo,

The only thing it does is pass along the server name to the Node
tls.connect() function.

t's not trivial to test that it actually works in a unit test since it
would require setting up a much more advanced web server / reverse proxy
with multiple certificates that would know what to do with the SNI
handshake.

Or maybe you have something else in mind?

**Comment by mcollina** [comment]
It's not just a matter of specifying a `SNICallback` when a server is created https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener?

File: README.md [hunk 1]
```diff
@@ -90,6 +90,8 @@ Available options:
         Print the output as newline delimited json. This will cause the progress bar and results not to be rendered. default: false.
   -f/--forever
         Run the benchmark forever. Efficiently restarts the benchmark on completion. default: false.
+  -s/--sniServerName
```
Comment (mcollina) [inline README.md]: I would prefer if this was `-S/--servername`.

**Comment by sebbarg** [comment]
Sure thing - capital S is already in use, so it has to be lowercase.

**Comment by sebbarg** [comment]
I will have a look

**Comment by sebbarg** [comment]
All done. The unit test tests whether the server name is sent correctly
through to tje https / tls connection. Since it is working on tcp level
rather than http, the test server constructs a http response with the
server name in the header. In that way the client side of the unit test can
reuse the http functionality to verify if the name was correctly sent
through.

**Pull request merged**